src/js/ConfigStorage.js

```diff
@@ -5,6 +5,9 @@
 var ConfigStorage = {
     // key can be one string, or array of strings
     get: function(key, callback) {
+      if (chrome.storage && chrome.storage.local) {
+          chrome.storage.local.get(key,callback);
+      } else {
```

Comment:
It must not be just the opposite? If windows.localStorage, then use the old code. If not, go back to the old chrome.storage. Or the nw builds did not have the chrome.storage?

Reply:
When I tried looking at window.localStorage in the Chrome APP it complained in the console. I didn't check to see if it'd complain every time I checked.

Reply:
Maybe its not a big problem, dev tools does collect them together...
![image](https://user-images.githubusercontent.com/190571/62612079-a0f02a80-b8bb-11e9-90ac-b26ffa943940.png)

Reply:
@Docteh:
- we already get other console error messages in the Chrome app (`history.length` is the same as yours, and it took me a lot of hacking away at the Google Analytics one to concede that this just isn't fixable;
- in your case, would checking `window.hasOwnProperty('localStorage')` be away to test without triggering the warning?

Reply:
```
window.hasOwnProperty('localStorage')
true
```

Looks like we're checking once and remembering the result.

Reply:
@Docteh:

> window.hasOwnProperty('localStorage')
> true

That's in the Chrome app?

Reply:
Yeah, Chrome App on windows. I can't check Chrome App on ChromeOS.

Reply:
lol, they added the function, only to make it spit out an error message.

We use `getNwGui()` in `src/js/main.js` to detect NW / Chrome in a centralised place. Do you think we should use this in this case as well?

Reply:
I would like to find a proper way to check if we're running as a Chrome App, as running a regular web page outside of a Chrome App wouldn't get a nwGui object either.

Reply:
Agreed - we probably need a split between NW.js, Chrome web app, and 'other'.

Reply:
how about in `src/js/main.js` we check if we don't have nwGui, but have `chrome.storage.local`? We could set something to false if that is the case.

Reply:
Sounds reasonable - or create an 'enum' for NW / Chrome / other.